src/services/stringCompletions.ts

```diff
@@ -1021,20 +1093,26 @@ function getCompletionsForPathMapping(
     packageDirectory: string,
     extensionOptions: ExtensionOptions,
     isExportsWildcard: boolean,
+    isImports: boolean,
+    compilerOptions: CompilerOptions,
     host: LanguageServiceHost,
+    moduleSpecifierResolutionHost: ModuleSpecifierResolutionHost | undefined
 ): readonly NameAndKind[] {
-    if (!endsWith(path, "*")) {

```

Comment:
This should also fix completions when `*` is somewhere in the middle of `paths`. The algorithm is largely the same between paths/exports/imports after all.

Note that nothing here fixes completions with multiple wildcards in the value - I don't want to work on that.